### PR TITLE
Added foldEffect from Signal that works on the UI type

### PR DESCRIPTION
**Pull request by KovaxG**
This is needed for functionality described in this issue: https://github.com/sharkdp/purescript-flare/issues/33

### CODE REVIEW

File: src/Flare.purs [hunk 1]
```diff
@@ -413,6 +415,15 @@ liftSF f (UI setup) = UI do
 foldp :: forall a b. (a -> b -> b) -> b -> UI a -> UI b
 foldp f x0 = liftSF (S.foldp f x0)
 
+-- | Creates a past dependent component with an effectful computation.
+-- | The function argument takes the value of the input component, and
+-- | the previous value of the output component, to produce the new value
+-- | of the output component wraped inside an Effect action.
```
Comment (sharkdp) [inline src/Flare.purs]: small typo: wrapped

**Comment by sharkdp** [comment]
Thank you very much for reporting this and for proposing a solution!

> You have a button that says "INC" and a number. If you press "INC" the number will increase. When the number hits 10, a new button will appear that says "RESET". If you press it the number will reset to 0, and the "RESET" button will disappear.

If you want and find the time, I think that would be a great example to add to the examples in `html/index.html` / `test/Main.purs`.

File: src/Flare.purs (hunk 1 above)
Comment (KovaxG) [reply src/Flare.purs]: got it, thanks

**Comment by KovaxG** [comment]
I would add the implementation, but the code is a bit messy, since I need stuff like channels and wrapped signals to do it. I think I should add the framework function I described in the issue. However that might be a bit of a weird function signature, do you think I should add it simply to Flare?

I'm talking about this:
```
play :: forall s i r. Eq s
          => s
          -> (s -> UI i)
          -> (i -> s -> s)
          -> (s -> r)
          -> Effect (UI r)
play s0 modIn update render = do
  stateChannel <- channel initialState
  pure $ map render
       $ foldEffect (\i -> (\s -> send stateChannel s $> s) <<< update i) s0
       $ innerFlare (wrap $ dropRepeats $ subscribe stateChannel) modIn
```

**Comment by sharkdp** [comment]
> However that might be a bit of a weird function signature, do you think I should add it simply to Flare?

Hm, probably not. But thank you for considering!

I'll merge this in the current state. Let me know if I should tag a new release.

**Pull request merged**